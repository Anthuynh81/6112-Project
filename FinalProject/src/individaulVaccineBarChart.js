import React, {useEffect, useState} from 'react';
import {Bar} from 'react-chartjs-2';
import Axios from 'axios';

function IndividualVaccineBarChart() {
    const [data, setData] = useState([]);

    useEffect(() => {
        Axios.get("http://localhost:5000/Vaccine").then((response) =>{
            setData(response.data)
        });
    }, []);

    const labels = ['Pfizer', 'Moderna', 'Johnson & Johnson'];


    let datapoints = [data[0].Pfizer, data[0].Moderna, data[0].JohnsonJohnson]

    const data1 = {
        labels,
        datasets: [
            {
                label: 'Vaccination',
                data: datapoints,
                backgroundColor: 'rgba(255, 99, 132, 0.5)',
            }
        ],
    };


    return (
        <div>
            <h2>Bar Chart</h2>
            <Bar data={data1} />
        </div>
    );
}

export default IndividualVaccineBarChart;