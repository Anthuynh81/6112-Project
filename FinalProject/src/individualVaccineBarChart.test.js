import Axios from 'axios';
import renderer from "react-test-renderer";
import individaulVaccineBarChart from "./individaulVaccineBarChart";
import IndividualVaccineBarChart from "./individaulVaccineBarChart";


async function getValue() {
    const response = await Axios.get("http://localhost:5000/Vaccine")
    return response
}

test("Get request", async () => {
    const response = await getValue()
    //Success code for get request
    expect(response.status).toEqual(200)
})

test("Getting correct data", async () => {
    const response = await getValue()
    // Should return USA vaccination information
    expect(response.data.Code).toBe("USA")
    // Should be empty as it should have no county information
    expect(response.data.Entity).toBe("United States")
})

test('returns the line chart when called', () => {
    const component = renderer.create(
        <IndividualVaccineBarChart />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
