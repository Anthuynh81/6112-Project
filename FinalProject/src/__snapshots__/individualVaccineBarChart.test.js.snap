// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns the line chart when called 1`] = `
<div>
  <h2>
    Bar Chart
  </h2>
  <canvas
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
